sympy/polys/matrices/domainmatrix.py

```diff
@@ -30,6 +30,19 @@ def __init__(self, rows, shape, domain):
         self.shape = shape
         self.domain = domain
 
+    def __getitem__(self, key):
+        i, j = key
+        m, n = self.shape
+        if not (isinstance(i, slice) or isinstance(j, slice)):
+            return DomainScalar(self.rep.getitem(i, j), self.domain)
+
+        if not isinstance(i, slice):
+            i = slice(i, i+1)
+        if not isinstance(j, slice):
+            j = slice(j, j+1)
```

Comment:
I don't think this will work if `i` or `j` is `-1`:
```python
In [116]: M = Matrix(range(9)).reshape(3, 3)

In [117]: M
Out[117]: 
⎡0  1  2⎤
⎢       ⎥
⎢3  4  5⎥
⎢       ⎥
⎣6  7  8⎦

In [118]: M[:, -1]
Out[118]: 
⎡2⎤
⎢ ⎥
⎢5⎥
⎢ ⎥
⎣8⎦

In [119]: M[:, -1:0]
Out[119]: []
```

Reply:
Sure, will fix it.
Thanks.